### DISCUSSION TITLE
Docker Symfony not working without internet connection

### DISCUSSION THREAD

**Discussion by dhodo** [symfony/symfony#43370] [Q&A]
Hi,

I am trying to use my Symfony docker setup on local and run properly as long as I have internet connection. At the time I disconnect from a wired/wifi connection, I am not able to use my Symfony local project anymore. It just stand still loading forever. This just happens since I use docker because before use it I could work on my local projects without internet connection. It happens to me even with a fresh new Symfony installation. Any idea why is this happening? This is my docker-compose config:

----------------------------
<pre><code>
version: '3.8'
services:
db:
image: mysql:5.7
    container_name: ${NAME}_sql
    volumes:
      - $PWD/mysql:/var/lib/mysql
      - $PWD/my.cnf:/etc/mysql/my.cnf
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USER: ${MYSQL_USER}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
    ports:
      - "3306:3306"
    stdin_open: true
    tty: true
  web:
    build: './'
    container_name: ${NAME}_web
    volumes:
      - ${PROJECT_FOLDER}/${NAME}:/var/www/html
      - $PWD/php.ini:/usr/local/etc/php/php.ini
    ports:
      - "80:80"
    stdin_open: true
    tty: true
  phpmyadmin:
    image: phpmyadmin/phpmyadmin
    container_name: ${NAME}_phpmyadmin
    links:
      - db:db
    ports:
      - "8082:80"
    environment:
      - PMA_USER=root
      - PMA_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - PHP_UPLOAD_MAX_FILESIZE=5000MB
  adminer:
    image: adminer:latest
    container_name: ${NAME}_adminer
    environment:
      ADMINER_DEFAULT_SERVER: db
    ports:
      - "8080:8080"
    volumes:
      - ~/adminer:/var/www/html
</code></pre>
----------------------------

Regards,
dhodo

**Comment by ghost** [comment]
Please format docker-compose config using markdown feature
```yml
key:
  subkey1: 'one'
  subkey2: 'two'
```

I think this is not an issue about Symfony but Docker and probably your skills in it.

**Reply by dhodo** [reply]
Sorry, I didn't knwo how to do it. Now it right.

I guess it's not about Symfony and more about how to define a docker-compose but I can't get it. Any idea? Do you see something wrong in my docker-compose file?

**Reply by michaoj** [reply]
Nothing seems to require an internet connection from what I see except to pull the images.

Do you still have the required images on your docker image store (`docker image ls`)

**Reply by dhodo** [reply]
Yep, I have all the images downloaded. It's kind of weird but the only way I tried to fix it it's using network_mode: host so the containers have the default ports. Anyway, I didn't fully fix it neither on this way.

**Comment by dhodo** [comment]
Does anyone know why I get connection refused as soon as I disconnect internet from my computer? There is nothing that requires internet connection to develop a Symfony project on a local machine with docker

**Comment by lyrixx** [comment]
> I'm able to work with Symfony + docker even without internet.

Maybe you have something in your application or a bundle that require internet.

You can try to put a `echo 'hello';exit;` in the front controller `public/index.php` to check if it comes from docker or from your application

**Reply by dhodo** [reply]
Hi! Thanks for you response

Definitely, the problem comes from docker because I echo any string and then exit but no luck. It seems the browser can't even connect to the container. I tried Firefox and Chrome (even in private mode and cleaning cache) but nothing works.

I can exec into the container at the command line when I have no internet connection, therefore the container is still available. I don't see anything different that may be relevant when comparing with other docker-compose.yml that I check on the internet.

Any idea of what may be happening? I use Manjaro.

Thanks!

**Reply by lyrixx** [reply]
No I don't but if you can create a reproducer I may help

**Reply by dhodo** [reply]
How could I do that?

**Reply by lyrixx** [reply]
You can push some code / config  to a public repository

**Reply by dhodo** [reply]
Great, I uploaded my config to github, this is the repo url:

https://github.com/dhodo/docker-reproducer

Try to reproduce it and tell me if it is working once you install everything and try to open Symfony with no Internet access in your computer. I think the problem it's not related with the docker-compose or Dockerfile configuration but something deeper that I have no idea what it is. It is clear that is not a Symfony-related problem because y echo a string and exit() at the index.php before anything and I still can't not access from the browser when I should see the string printed on it. I tried on a Symfony fresh installation, no new vendor installed.

Thanks for your help!

**Reply by lyrixx** [reply]
I tried an everything is working fine. 

I see nothing wrong in your configuration.

Maybe you have an issue with your local DNS, try hitting 127.0.0.1 or the direct container IP for debugging. 

Unfortunately, this is not related with Symfony and I cannot help you more with this issue.

**Reply by dhodo** [reply]
Okay! Knowing that the docker conf is fine has been already a great help. I'll check my local DNS. Thank you so much!
